nexus/src/app/rack.rs

```diff
@@ -724,4 +716,51 @@ impl super::Nexus {
 
         Ok(result)
     }
+
+    /// Return the list of sleds that are inserted into an initialized rack
+    /// but not yet initialized as part of a rack.
+    pub(crate) async fn uninitialized_sled_list(
+        &self,
+        opctx: &OpContext,
+    ) -> ListResultVec<UninitializedSled> {
+        // Grab the SPs from the last collection
+        let limit = NonZeroU32::new(50).unwrap();
+        let collection = self
+            .db_datastore
+            .inventory_get_latest_collection(opctx, limit)
+            .await?;
+        let pagparams = DataPageParams {
+            marker: None,
+            direction: dropshot::PaginationOrder::Descending,
+            // TODO: This limit is only suitable for a single sled cluster
+            limit: NonZeroU32::new(32).unwrap(),
```

Comment:
Same here

Reply:
I started to do this, but we can safely hit the limit of 32 here. I could change this to be 33 so that we error on multi-rack I suppose, but by that point we'll have many other issues in testing.